libs2ecore/src/S2EExecutor.cpp

```diff
@@ -1394,32 +1407,40 @@ S2EExecutor::StatePair S2EExecutor::fork(ExecutionState &current, const klee::re
         currentState->forkDisabled = true;
     }
 
-    res = Executor::fork(current, condition, keepConditionTrueInCurrentState);
+    if (condition) {
+        res = Executor::fork(current, *condition, keepConditionTrueInCurrentState);
+    } else {
+        res = Executor::fork(current);
+    }
 
     currentState->forkDisabled = oldForkStatus;
 
     if (!(res.first && res.second)) {
         return res;
     }
 
-    S2EExecutionState *newStates[2];
+    std::vector<S2EExecutionState *> newStates(2);
+    std::vector<klee::ref<Expr>> newConditions(2);
```

Comment:
Use `llvm::SmallVector<..., 2>` instead to minimize memory allocations.

Reply:
Thanks!

Actually, there are also `std::vector`s in `S2EExecutor::notifyFork()`, maybe they can be replaced as well (probably in a separate PR)?